pilot/pkg/model/config_status.go

```diff
@@ -0,0 +1,91 @@
+package model
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// config_status tracks configuration metrics and errors.
+// Metrics are reset after a push - at the beginning all values are zero, and when push completes the status
+// is reset.
+
+var (
+
+	// LastPushStatus preserves the metrics and data collected during lasts global push.
+	// It can be used by debugging tools to inspect the push event. It will be reset after each push with the
+	// new version.
+	LastPushStatus *PushStatus
+)
+
+var (
+	ipNotFound = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "pilot_no_ip",
+		Help: "Pods not found in the endpoint table, possibly invalid.",
+	})
+
+	// TODO: use the cluster ID or parse the proxyID to determine the workload names and namespaces, for better reporting
+	//ipNotFound = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	//	Name: "pilot_no_ip",
+	//	Help: "Pods not found in the endpoint table, possibly invalid.",
+	//}, []string{"workload"})
+
+	unready = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "pilot_unready",
+		Help: "Endpoints found in unready state",
+	})
+
+	conflictingOutbound = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "pilot_conf_out_listeners",
+		Help: "Number of conflicting listeners.",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(ipNotFound)
+	prometheus.MustRegister(unready)
+	prometheus.MustRegister(conflictingOutbound)
+
+}
+
+type PushStatus struct {
```

Comment:
Let's not expose the mutex .... perhaps all of the fields should be private. Instead, add methods that do the appropriate locking.

Reply:
I can do it in a follow-up, right now priority is to verify it works and get the basics out. 
There are few ways to make it more generic and avoid the mutex, but can wait for 1.1.

Reply:
I don't see a benefit at this point - will try to add an interface, but right now there are far bigger problems ( detecting when a push is over, the fact changes may happen independent of pushes, etc). Which may require even bigger changes to the interface. Once  the basic infra works - and before we convert all warnings and errors to metrics - I'll have a better interface.

Reply:
Sure that's fine, but I still expect this cleanup to be part of this PR before submission.